### PR TITLE
@W-16814802: [iOS] Add Deep Link From External QR Code Reader To iOS Native Swift Template QR Code Login

### PR DESCRIPTION
**Pull request by JohnsonEricAtSalesforce**
🤘🏻 _*Ready For Review!*_ 🎸

  This is a parallel to https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2625.  The crux of the change for MSDK is updating the naming conventions, documentation and a few customization options to match the public API for QR Code Log In we just finished for Android.

  Thanks, as always, for reading along ⭐️

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController+QrCodeLogin.swift [hunk 1]
```diff
@@ -28,44 +28,25 @@
...
-     *     - loginQrCodeContent:  The login QR code content.  This should be either a URL or URL
-     *     query containing the UI Bridge API JSON parameter.  The UI Bridge API JSON parameter
-     *     should contain URL-encoded JSON with two values:
-     *       - frontdoor_bridge_url
-     *       - pkce_code_verifier
-     * If pkce_code_verifier is not specified then the user agent flow is used
-     * - Returns: Boolean true if a log in attempt is possible using the provided QR code content, false
-     * otherwise
-     */
-    func loginFromQrCode(
```
Comment (JohnsonEricAtSalesforce) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController+QrCodeLogin.swift:46]: I just moved this method one down so it was easier to read side by side with the Android versions. 😉

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController+QrCodeLogin.swift [hunk 2]
```diff
@@ -28,44 +28,25 @@
...
-        }
-    }
-
+    
+    // MARK: - QR Code Login Via Salesforce Identity API UI Bridge Public Implementation
+    
     /**
      * Automatically log in with a UI Bridge API front door bridge URL and PKCE code verifier.
+     *
+     * This method is the intended entry point to Salesforce Mobile SDK when using the Salesforce
```
Comment (JohnsonEricAtSalesforce) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController+QrCodeLogin.swift:37]: I pulled the documentation along to match the Android implementation.  It's very similar reading, so I really tried to match sure I updated all the terminology to iOS SDK context.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController+QrCodeLogin.swift [hunk 3]
```diff
@@ -28,44 +28,25 @@
...
+     * entry point is used to log in with the front door URL and optional PKCE code verifier.
+     *
      * - Parameters
      *   - frontdoorBridgeUrl: The UI Bridge API front door bridge URL
-     *   - pkceCodeVerifier: The PKCE code verifier
+     *   - pkceCodeVerifier: The optional PKCE code verifier, which is not required for User Agent
+     * Authorization Flow but is required for Web Server Authorization Flow
      */
-    func loginWithFrontdoorBridgeUrl(
+    class func loginWithFrontdoorBridgeUrl(
```
Comment (JohnsonEricAtSalesforce) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController+QrCodeLogin.swift:49]: This methods do live with the login activity on Android, but I hadn't moved them all to its (static) companion object.  This was convenient for allowing the app's scene delegate to call in, but perhaps we want to scope them in a different way?  I'm open to suggestions or this works.

**Review by JohnsonEricAtSalesforce — COMMENTED** [review]
Here's a few self-review notes to start the conversation.

**Review by bbirman — APPROVED** [review]

**Review by wmathurin — APPROVED** [review]

**Pull request merged**